### ISSUE TITLE
How to use Single Mode when I am training for the same graph structure

### ISSUE THREAD

**Issue by a-sarabi** [danielegrattarola/spektral#439]
I'm trying to implement a Graph Attention layer (GATConv), but I keep manually duplicating the same adjacency matrix for each instance using np.tile(). This leads to memory issues, especially when dealing with large benchmark datasets. Is there a way to streamline the process and pass just one adjacency matrix to the model?

def build_model(input_shape, correlation_shape, use_graph_layer):
.......
 x = tf.transpose(inputs, perm=[0, 2, 1])
                do_1 = Dropout(dropout)(x)
                gc_1 = GATConv(
                    channels,
                    attn_heads=n_attn_heads,
                    concat_heads=True,
                    dropout_rate=dropout,
                    activation="elu",
                    kernel_regularizer=l2(l2_reg),
                    attn_kernel_regularizer=l2(l2_reg),
                    bias_regularizer=l2(l2_reg),
                )([do_1, inp_lap])
.........

history = model.fit(
                [X_train, **np.tile(correlation_train, (len(X_train), 1, 1))**, decoder_Y_train],
                Y_train,
                validation_split=0.2,
                epochs=100,
                callbacks=callbacks,
                batch_size=50
            )
            
Thank you in Advance

**Comment by a-sarabi** [comment]
I solved this issue by using a data_generator function to train per batch size.
